apps/datahub/src/app/dataset/dataset-information/dataset-information.component.ts

```diff
@@ -31,4 +41,10 @@ export class DatasetInformationComponent {
 
     return categoryKeywords?.map((keyword) => keyword.label)
   }
+
+  onInfoKeywordClick(keyword: string) {
+    this.routerFacade
+      ? this.routerFacade.updateSearch({ q: keyword })
```

Comment:
You are right that the ticket states to implement a filter based on `q=`. However, this does not synchronize the category filter drop down. I think we already discussed about this when we estimated the work, but I can't remember the outcome @jahow. Also, not sure if the change that we now display keywords from a thesaurus here instead of `topics`, came after this discussion and if it makes things more complicated.

Reply:
You can try `categoryKeyword=...` and see if that shows up correctly

Reply:
Hum I have the impression that there is still the issue that when clicking on the category badge we use the `label` (eg. http://localhost:4200/search?categoryKeyword=Environnement,%20%C3%A9nergie) while the filter drop down uses the `key` (eg. http://localhost:4200/search?categoryKeyword=https:%2F%2Fdata.lillemetropole.fr%2Fthematique%2Fcategories%2Fenvironnement_energie) which is returned in the aggregation. Debugging, I don't see the key being part of the category [here](https://github.com/camptocamp/mel-dataplatform/blob/filter-by-category-ds-page/apps/datahub/src/app/dataset/dataset-information/dataset-information.component.ts#L41) though. Do you think we need further modifications in gn-ui @jahow ?

Reply:
I think it should be straightforward to filter on the key instead of label in that component; the `Keyword` model definitely has the key in it, so it can probably be used

Reply:
I am unsure about what it means I should do... Should I change categoryKeywork.label to categoryKeyword.key [here](https://github.com/camptocamp/mel-dataplatform/blob/filter-by-category-ds-page/apps/datahub/src/app/dataset/dataset-information/dataset-information.component.ts#L41) then?

Reply:
I think we'd need to return the entire object here, to display the `label` in the html, but pas the `key` to `routerFacade.updateSearch`. I had the impression that the key wasn't present here though, looking earlier. You may have to dig deeper why, if it's the case.

Reply:
I've opened https://github.com/geonetwork/geonetwork-ui/pull/937 in gn-ui to add the key. This should make your added commits work @cmoinier. @jahow could you take a look at the PR I opened when you come back if other changes are necessary ? Thanks :-)